tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -0,0 +1,214 @@
+            return;
+        }
+
+        System.out.println("Resetting offsets to zero for topics " + sourceTopics);
+
+        final Properties config = new Properties();
+        config.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, this.options.valueOf(bootstrapServerOption));
+        config.setProperty(ConsumerConfig.GROUP_ID_CONFIG, this.options.valueOf(applicationIdOption));
+        config.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+        final KafkaConsumer<byte[], byte[]> client = new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+        try {
+            client.subscribe(sourceTopics);
+            client.poll(1);
+
+            for (final TopicPartition partition : client.assignment()) {
+                client.seek(partition, 0);
+            }
+            client.commitSync();
+        } catch (final RuntimeException e) {
+            System.err.println("Resetting source topic offsets failed.");
+            throw e;
+        } finally {
+            client.close();
+        }
+    }
+
+    private void seekToEndIntermediateTopics(final ZkUtils zkUtils, final List<String> allTopics) {
+        if (this.options.has(intermediateTopicsOption)) {
+            System.out.println("Seek-to-end for intermediate user topics.");
+
+            final Properties config = new Properties();
+            config.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, this.options.valueOf(bootstrapServerOption));
+            config.setProperty(ConsumerConfig.GROUP_ID_CONFIG, this.options.valueOf(applicationIdOption));
+            config.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+            for (final String topic : this.options.valuesOf(intermediateTopicsOption)) {
+                if (allTopics.contains(topic)) {
+                    System.out.println("Topic: " + topic);
+
+                    final KafkaConsumer<byte[], byte[]> client = new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+                    try {
+                        client.subscribe(Collections.singleton(topic));
+                        client.poll(1);
+
+                        final Set<TopicPartition> partitions = client.assignment();
+                        client.seekToEnd(partitions);
+                        client.commitSync();
+                    } catch (final RuntimeException e) {
+                        System.err.println("Seek to end for topic " + topic + " failed");
+                        throw e;
+                    } finally {
+                        client.close();
+                    }
+                }
+            }
+
+            System.out.println("Done.");
+        } else {
+            System.out.println("No intermediate topics specified.");
+        }
+    }
+
+    private void deleteInternalTopics(final ZkUtils zkUtils, final List<String> allTopics) {
+        System.out.println("Deleting internal topics.");
```

Comment:
Nit: "Deleting any topics used internally by Kafka Streams for the application with id [...]" (or sth like that)